Extension/lib/start.js

```diff
@@ -17,44 +17,60 @@
 
 (function () {
 
-    adguard.console.info('Starting adguard... Version: {0}. Id: {1}', adguard.app.getVersion(), adguard.app.getId());
-
-    // Initialize popup button
-    adguard.browserAction.setPopup({
-        popup: adguard.getURL('pages/popup.html')
-    });
-
-    /**
-     * Start application
-     */
-    adguard.filters.start({
-
-        onInstall: function (callback) {
-
-            // Process installation
-
-            /**
-             * Show UI installation page
-             */
-            adguard.ui.openFiltersDownloadPage();
-
-            /**
-             * Tracking extension install or update according to http://adguard.com/en/privacy.html#browsers
-             * We do this with a single purpose: to know the number of unique installations of our extension.
-             * This information is stored for 24 hours and then it is deleted.
-             *
-             * The only thing which is not deleted is the aggregated info: installs count and active users count.
-             */
-            adguard.backend.trackInstall();
-
-            // Retrieve filters and install them
-            adguard.filters.offerFilters(function (filterIds) {
-                adguard.filters.addAndEnableFilters(filterIds, callback);
-            });
+    function preInitialize(callback) {
```

Comment:
The only purpose of this function is to somehow sneak the migration call to the extension initialization.

Instead of the vague hooks mechanics, we'd better use a more straightforward approach.

Let's have a `startup` module here:
```
adguard.initialize = function() {};
```

And let's have a browser-specific `start.js`, which will call `adguard.initialize` when it is a right time.